### PR TITLE
Update report to 2018-03-13

### PR DESCRIPTION
**Pull request by hollowbot**
# Report for tests performed on 2018-03-13

### Security Headers

| Test  | First View | Repeat View |
| ----- | ---------- | ----------- |
| Grade | A          | N/A         |

### [WebPageTest](http://www.webpagetest.org/results.php?test=180313_FK_8f0e1e51558a0530678658d281b1e05a)

| Test                       | First View | Repeat View |
| -------------------------- | ---------- | ----------- |
| Number of requests         | 58         | 46          |
| Time to first byte         | 1.78s      | 1.79s       |
| Fully loaded               | 43.50s     | 32.75s      |
| Response size              | 739.88KiB  | 597.12KiB   |
| Response size (compressed) | 380.63KiB  | 440.73KiB   |

### Security Headers

| Test  | First View | Repeat View |
| ----- | ---------- | ----------- |
| Grade | A          | N/A         |

### [WebPageTest](http://www.webpagetest.org/results.php?test=180313_F1_eebd7df58bdb7c778053b6cb5a43cb12)

| Test                       | First View | Repeat View |
| -------------------------- | ---------- | ----------- |
| Number of requests         | 16         | 3           |
| Time to first byte         | 1.84s      | 1.75s       |
| Fully loaded               | 4.43s      | 2.54s       |
| Response size              | 201.86KiB  | 7.13KiB     |
| Response size (compressed) | 158.80KiB  | 0.00KiB     |

### Lighthouse via WebPageTest

| Test                     | First View | Repeat View |
| ------------------------ | ---------- | ----------- |
| Performance              | 84%        | 84%         |
| First Meaningful Paint   | 2.38s      | 2.38s       |
| First Interactive        | 4.75s      | 4.75s       |
| Consistently Interactive | 4.75s      | 4.75s       |
| Speed Index Metric       | 3651       | 3651        |
| Estimated Input Latency  | 0.02s      | 0.02s       |
| PWA                      | 45%        | 45%         |
| Accessibility            | 97%        | 97%         |
| Best Practices           | 88%        | 88%         |
| SEO                      | 90%        | 90%         |

### CODE REVIEW

**Comment by wholesomedev** [comment]
@forabi Interesting best practices errors here? http://www.webpagetest.org/lighthouse.php?test=180313_F1_eebd7df58bdb7c778053b6cb5a43cb12&run=1#best-practices

**Comment by forabi** [comment]
Also look at the screenshots in the Lighthouse report, the styles are
missing. I'll look into that today.

**Comment by forabi** [comment]
I think we also need to handle `/?branch=` in `lambda-assign-env`.
(This is not user facing)

**Comment by wholesomedev** [comment]
Cool 👍

Another thing we should check is if these errors are getting logged to
SumoLogic, and if not, then can we log them to SumoLogic?
-- 
Sent from my iPhone

**Comment by forabi** [comment]
I don't think these can be logged because they are browser errors, not JS
errors.

**Comment by forabi** [comment]
Hmm, actually I no longer think that handling `/branch?` in `assign-env` will make any difference.

* I've checked the cache behaviors in CloudFront and we are correctly whitelisting both `branch` and `env` cookies as well as the `branch` query string parameter.
* This means that the `branch` parameter is always forwarded to Release Manager
* In Release Manager, `branch` always takes precedence over `env`, and we correctly send the header `Set-Cookie: branch=<branch>` in the response, which means that all subsequent request will be pointed to the specified branch. There is no chance that a request is redirected to an environment which does not have the static, versioned styles and scripts.

TL;DR: I don't know what's going on. This should not happen. :sweat: 

By the way, we should not be testing this URL because URLs with `branch` are not cached in CF. It does not reflect the real-world performance of the website. We should set the `env` cookie instead (for separate tests for `master` and `beta`). I don't know if we can do that in WebPageTest though. Or maybe we should just remove `?/branch`.

**Comment by wholesomedev** [comment]
@forabi If this issue has no user impact, it doesn't seem urgent. We can ignore it for now. But removing `brach=master` from the monitor could lead to these tests running on the `beta` branch occasionally. I'll go ahead and create an issue for looking into ensuring that the tests run against `master`.

**Comment by forabi** [comment]
> If this issue has no user impact, it doesn't seem urgent.

@wholesomedev I'm not sure, I haven't seen this issue before. Removing `branch` from the URL will allow us to check whether it happens for public pages or not.

> But removing brach=master from the monitor could lead to these tests running on the beta branch occasionally.

@wholesomedev I think WebPageTest sets (or allows us to set) a custom user agent, we can treat that like we treat bots in `lambda-assign-env`.
